File: 2021/2/partTwo.ts
import { readFileSync } from "fs";

async function main() {
  let result = 0;
  const file = readFileSync(process.argv[2], "utf-8");
  const rows: Array<[string, number]> = file.split("\n").map((r) => {
    const row = r.split(" ");
    return [row[0], parseInt(row[1])]
  });

  let hp = 0
  let vp = 0 
  let aim = 0

  rows.map(([instruction, amount], index) => {
    if(instruction === "forward") {
      hp += amount
      vp += aim * amount
    }
    if(instruction === "down") {
      // vp += amount # part 1
      aim += amount
    }
    if(instruction === "up") {
      // vp -= amount # part 1
      aim -= amount
    }
  });

  result = vp * hp;

  console.log(result);
}

main();

File: 2021/2/part-two.py
import sys

if __name__ == "__main__":
    with open(sys.argv[1], encoding="utf-8") as f:
        _rows = [x.split() for x in f]
        rows = []
        for row in _rows:
            i, a = row[0], int(row[1])
            rows.append((i,a))
        
    hp = 0
    depth = 0
    aim = 0

    for row in rows:
        i, a = row[0], row[1]
        if(i == "forward"):
            hp += a
            depth += aim * a
        if(i == "down"):
            aim += a
        if(i == "up"):
            aim -= a

    result = hp * depth

    print(result)